lib/pg_query/parse.rb

```diff
@@ -273,6 +273,7 @@ def load_objects! # rubocop:disable Metrics/CyclomaticComplexity
           when :join_expr
             from_clause_items << { item: next_item[:item].join_expr.larg, type: next_item[:type] }
             from_clause_items << { item: next_item[:item].join_expr.rarg, type: next_item[:type] }
+            subselect_items.concat << next_item[:item].join_expr.quals
```

Comment:
Oh, I just realized we are missing a `.to_ary` here (because of https://bugs.ruby-lang.org/issues/18140, per the comment on top of the function) -- I'll make a quick commit to adjust that